contracts/security/Initializable/InitializableStorage.sol

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.8;
+
+library InitializableStorage {
+    struct Layout {
+        uint8 initialized;
```

Comment:
I think it would be more gas-efficient to use a `uint256` - smaller uint types generally are only more efficient when they can be packed in storage.  I believe the `bool initializing` uses a full 32-bytes storage slot, so packing does not occur.

Reply:
Though, if `initializing` were converted to something like a `bytes1`, then I believe packing would occur.

Reply:
Nope, uint8 and bool will be packed into a single slot. `uint8` and `bool` are both 1 byte.
I think uint8 can change to uint248 to full a slot.

BTW, `bool` uses a full slot only when `mapping(uint256 => bool)`, due to how mapping decide which slot to store.

Reply:
I must have misunderstood - I read somewhere that `bool` takes more space than one would expect, but that presumed expectation must have been a single bit.